src/main/kotlin/org/rust/ide/intentions/CreateFunctionIntention.kt

```diff
@@ -0,0 +1,127 @@
+    val callExpr = ctx.callExpr
+
+    val factory = RsPsiFactory(project)
+    val config = getFunctionConfig(callExpr)
+
+    val genericParams = config.genericConstraints.buildTypeParameters()
+    val params = config.parameters.joinToString(", ")
+    val whereClause = config.genericConstraints.buildWhereClause()
+
+    val vis = if (callExpr.containingMod != ctx.module) "pub(crate) " else ""
+    val function = factory.tryCreateFunction("${vis}fn $functionName$genericParams($params)$whereClause {\n    unimplemented!()\n}")
+        ?: return
+
+    val sourceFunction = callExpr.parentOfType<RsFunction>() ?: return
+    val inserted = insertFunction(ctx.module, sourceFunction, function)
+
+    if (ctx.module.containingFile == callExpr.containingFile) {
+        val toBeReplaced = inserted.valueParameters.flatMap { listOfNotNull(it.pat, it.typeReference) } +
+            listOfNotNull(inserted.block?.expr)
+        editor.buildAndRunTemplate(inserted, toBeReplaced.map { it.createSmartPointer() })
+    } else {
+        // template builder cannot be used for a different file
+        inserted.navigate(true)
+    }
+}
+
+private fun getTargetModuleForFunction(path: RsPath): RsMod? {
+    if (path.qualifier != null) {
+        val mod = path.qualifier?.reference?.resolve() as? RsMod
+        if (mod?.containingCargoPackage?.origin != PackageOrigin.WORKSPACE) return null
+        if (!isUnitTestMode && !mod.isWritable) return null
+        return mod
+    }
+    return path.containingMod
+}
+
+private data class FunctionConfig(val parameters: List<String>,
+                                  val genericConstraints: GenericConstraints)
+
+private fun getFunctionConfig(callExpr: RsCallExpr): FunctionConfig {
+    val parameters = callExpr.valueArgumentList.exprList.mapIndexed { index, expr ->
+        "p$index: ${expr.type.renderInsertionSafe(useAliasNames = true)}"
+    }
+
+    val genericConstraints = GenericConstraints.create(callExpr)
+        .filterByTypes(callExpr.valueArgumentList.exprList.map { it.type })
+
+    return FunctionConfig(parameters, genericConstraints)
+}
+
+private fun insertFunction(
+    targetModule: RsMod,
+    sourceFunction: RsFunction,
+    function: RsFunction
+): RsFunction {
+    if (targetModule == sourceFunction.containingMod) {
+        val impl: RsTraitOrImpl? = when (val owner = sourceFunction.owner) {
+            is RsAbstractableOwner.Trait -> owner.trait
+            is RsAbstractableOwner.Impl -> owner.impl
+            else -> null
+        }
+
+        val target = impl ?: sourceFunction
+        return target.parent.addAfter(function, target) as RsFunction
+    }
+
+    // add to the end of module/file
+    return if (targetModule is RsModItem) {
+        targetModule.addBefore(function, targetModule.rbrace)
+    } else {
+        if (targetModule.lastChild == null) {
+            targetModule.add(function)
+        } else {
+            targetModule.addBefore(function, targetModule.lastChild)
```

Comment:
Hmm, before last child. This means the function will be the penultimate one, isn't it?

Reply:
Good catch, thanks. One of the tests was incorrect, so it didn't catch this.